examples/src/main/java/kafka/examples/Consumer.java

```diff
@@ -16,52 +16,52 @@
  */
 package kafka.examples;
 
-
-import java.nio.ByteBuffer;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.Collections;
 import java.util.Properties;
-import kafka.consumer.ConsumerConfig;
-import kafka.consumer.KafkaStream;
-import kafka.javaapi.consumer.ConsumerConnector;
-import kafka.message.MessageAndMetadata;
 
+import kafka.utils.ShutdownableThread;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
 
-public class Consumer extends Thread
+public class Consumer extends ShutdownableThread
 {
-  private final ConsumerConnector consumer;
+  private final KafkaConsumer<Integer, String> consumer;
   private final String topic;
-  
+
   public Consumer(String topic)
   {
-    consumer = kafka.consumer.Consumer.createJavaConsumerConnector(
-            createConsumerConfig());
+    super("KafkaConsumerExample", false);
+    Properties props = new Properties();
+    props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
+    props.put(ConsumerConfig.GROUP_ID_CONFIG, "DemoConsumer");
+    props.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "true");
+    props.put(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG, "1000");
+    props.put(ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, "30000");
+    props.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.IntegerDeserializer");
+    props.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.StringDeserializer");
+
+    consumer = new KafkaConsumer<>(props);
     this.topic = topic;
   }
 
-  private static ConsumerConfig createConsumerConfig()
-  {
-    Properties props = new Properties();
-    props.put("zookeeper.connect", KafkaProperties.zkConnect);
-    props.put("group.id", KafkaProperties.groupId);
-    props.put("zookeeper.session.timeout.ms", "400");
-    props.put("zookeeper.sync.time.ms", "200");
-    props.put("auto.commit.interval.ms", "1000");
-
-    return new ConsumerConfig(props);
+  @Override
+  public void doWork() {
+    consumer.subscribe(Collections.singletonList(this.topic));
+    ConsumerRecords<Integer, String> records = consumer.poll(1000);
+    for (ConsumerRecord<Integer, String> record : records) {
+      System.out.println("Received message: (" + record.key() + ", " + record.value() + ") at offset " + record.offset());
+    }
+  }
 
+  @Override
+  public String name() {
+    return null;
```

Comment:
This is because we are inheriting a Scala class, and moving forward we should remove this with the Java version 

```org.apache.kafka.copycat.util.ShutdownableThread``` that can be moved into ```o.a.k.common.utils```.

We can leave it as is for now.